packages/application-shell/src/components/application-shell/application-shell.spec.js

```diff
@@ -633,3 +639,596 @@ describe('when dispatching a loading notification', () => {
+      operations.FetchLoggedInUser.me.defaultProjectKey
+    );
+  });
+  it('should render links with all the correct state attributes', async () => {
+    const navbarSubmenuMock = ApplicationNavbarSubmenuMock.build();
+    const navbarMock = ApplicationNavbarMenuMock.build({
+      submenu: [navbarSubmenuMock],
+    });
+    const rendered = renderApp(null, {
+      featureFlags: {
+        [PROJECT_EXTENSIONS]: false,
+      },
+      DEV_ONLY__loadNavbarMenuConfig: () => Promise.resolve([navbarMock]),
+    });
+    // Get the nav container, to narrow down the search area
+    const container = await rendered.findByTestId('left-navigation');
+    const navbarRendered = within(container);
+
+    const applicationLocale = operations.FetchLoggedInUser.me.language;
+    const mainMenuLabel = navbarMock.labelAllLocales.find(
+      localized => localized.locale === applicationLocale
+    );
+    const mainSubmenuLabel = navbarSubmenuMock.labelAllLocales.find(
+      localized => localized.locale === applicationLocale
+    );
+    // Check the relationships between the menu items of a group
+    const menuTitle = await navbarRendered.findByText(mainMenuLabel.value);
+    const groupId = menuTitle.getAttribute('aria-owns');
+    expect(rendered.container.querySelector(`#${groupId}`)).toHaveAttribute(
+      'aria-expanded',
+      'false'
+    );
+    fireEvent.click(menuTitle);
+    expect(rendered.container.querySelector(`#${groupId}`)).toHaveAttribute(
+      'aria-expanded',
+      'true'
+    );
+    const menuGroup = within(rendered.getByTestId(groupId));
+    expect(menuGroup.queryByText(mainSubmenuLabel.value)).toBeInTheDocument();
+    expect(menuGroup.queryByText(mainSubmenuLabel.value)).not.toHaveAttribute(
+      'aria-current'
+    );
+    // Go to the link to check if the link becomes active
+    fireEvent.click(menuGroup.queryByText(mainSubmenuLabel.value));
+    await wait(() => {
+      expect(menuGroup.queryByText(mainSubmenuLabel.value)).toHaveAttribute(
+        'aria-current',
+        'page'
+      );
+    });
+  });
+});
+describe('when rendering navbar menu links from remote config', () => {
+  let mcOperations;
+  let proxyOperations;
+  beforeEach(() => {
+    mcOperations = mocksForMc.createMockOperations();
+    proxyOperations = mocksForProxy.createMockOperations();
+    createGraphqlMockServer(xhrMock, {
+      operationsByTarget: { mc: mcOperations, proxy: proxyOperations },
+    });
+    selectProjectKeyFromUrl.mockReturnValue(
+      mcOperations.FetchLoggedInUser.me.defaultProjectKey
+    );
+  });
+  it('should render links with all the correct state attributes', async () => {
+    const rendered = renderApp(null, {
+      featureFlags: {
+        [PROJECT_EXTENSIONS]: false,
+      },
+      environment: {
+        servedByProxy: 'true',
+      },
+    });
+    // Get the nav container, to narrow down the search area
+    const container = await rendered.findByTestId('left-navigation');
```

Comment:
Nit: maybe naming container a bit better would help understanding where we are in the DOM. Like `leftNavigationContainer` or something?